packages/block-editor/src/components/inserter/index.js

```diff
@@ -161,11 +161,14 @@ export default compose( [
 			getBlockRootClientId,
 			hasInserterItems,
 			__experimentalGetAllowedBlocks,
+			getBlockSelectionEnd,
 		} = select( 'core/block-editor' );
 		const { getBlockVariations } = select( 'core/blocks' );
 
 		rootClientId =
-			rootClientId || getBlockRootClientId( clientId ) || undefined;
+			rootClientId ||
+			getBlockRootClientId( clientId || getBlockSelectionEnd() ) ||
```

Comment:
#23263 proposes to revert this particular changes, as it caused the default root appender to insert blocks wherever selection is.

Reply:
Seems logical @talldan the widget screen should  inject the rootClientId as a prop of the inserter instead.